Sources/ExportKit/Views/ExportViewModel.swift

```diff
@@ -41,6 +41,59 @@ class ExportViewModel: ObservableObject {
             alert = .descriptiveError(error.localizedDescription)
         }
     }
+
+    func exportAll() {
+        guard let exportStrategy = ExportKit.shared.exportStrategy else {
+            alert = .descriptiveError("No export strategy")
+            ExportKit.shared.logHandler?("[ EXPORTER ] No export strategy")
+            return
+        }
+
+        guard let topViewController = UIApplication.topViewController() else {
+            alert = .descriptiveError("No Top View Controller")
+            ExportKit.shared.logHandler?("[ EXPORTER ] No Top View Controller")
+            return
+        }
+
+        let results = data.reduce(.success([Item]())) { (partialResult, item) -> Result<[Any], Error> in
+            switch partialResult {
+            case .failure(_):
+                return partialResult
+            case .success(let array):
+                switch exportStrategy(item) {
+                case .success(let itemResult):
+                    return .success(array + CollectionOfOne(itemResult))
+                case .failure(let error):
+                    return .failure(error)
+                }
+            }
+        }
+
+        switch results {
+        case .success(let items):
+            var itemsToShare = [Any]()
+
+            if let textItems = items as? [String] {
+                let textToShare = textItems.reduce("") { partialResult, item in
+                    partialResult == "" ? item : partialResult + "\n" + item
```

Comment:
what do you think of simplifying this line to something like:
```
if partialResult.isEmpty {
    return item 
} else {
    return partialResult + "\n" + item
}
```

Reply:
This is much better! I pushed it 👍